packages/api/lib/rulesHelpers.js

```diff
@@ -104,6 +104,44 @@ function lookupCollectionInEvent(eventObject) {
   });
 }
 
+/*
+ * Build payload from rule for lambda invocation
+ *
+ * @param {ApiRule} rule              - API rule
+ * @param {object} [cumulusMeta]      - Optional cumulus_meta object
+ * @param {string} [asyncOperationId] - Optional ID for asynchronous operation
+ *
+ * @returns {Object} lambda invocation payload
+ */
+async function buildPayload(rule, cumulusMeta, asyncOperationId) {
+  // makes sure the workflow exists
+  const bucket = process.env.system_bucket;
```

Comment:
We need to be testing to make sure that these two environment variables are set and doing something if they are. Probably throw an exception?

Reply:
Hmm I do agree that we should be defensive but I'm not sure you could do much of anything if system_bucket or stackName weren't set. They're required for deploying pretty much all of our modules. Validating that they exist is probably a good idea as I'm not sure where it fails if you don't provide them but I'm not sure that this is the right spot. It feels pretty late in the process if we're gonna catch it. I guess we could catch it here because it's easy but it feels a little random? Just me?

Also, heads up that this function isn't new. It was just moved in the file. We should still obvious make it better.